src/Core/Bridge/Elasticsearch/DataProvider/CollectionDataProvider.php

```diff
@@ -77,10 +89,18 @@ public function supports(string $resourceClass, ?string $operationName = null, a
             return false;
         }
 
-        try {
-            $this->identifierExtractor->getIdentifierFromResourceClass($resourceClass);
-        } catch (NonUniqueIdentifierException $e) {
-            return false;
+        if ($this->identifierExtractor) {
+            try {
+                $this->identifierExtractor->getIdentifierFromResourceClass($resourceClass);
+            } catch (NonUniqueIdentifierException $e) {
+                return false;
+            }
+        } else if ($this->resourceMetadataFactory instanceof ResourceMetadataCollectionFactoryInterface) {
```

Comment:
```suggestion
        } elseif ($this->resourceMetadataFactory instanceof ResourceMetadataCollectionFactoryInterface) {
```

Reply:
And what happens if there is no identifiers extractor and no resource metadata factory?

Reply:
You are right, this condition is unnecessary, the check is already done inside the constructor.